packages/ketcher-core/src/domain/entities/monomer-chains/Chain.ts

```diff
@@ -71,7 +84,21 @@ export class Chain {
     if (!monomer) return;
 
     this.add(monomer);
-    this.fillSubChains(getNextMonomerInChain(this.lastNode?.lastMonomerInNode));
+    if (this.lastNode instanceof Nucleotide) {
+      this.fillSubChains(
+        getNextMonomerInChain(
+          getPhosphateFromSugar(monomer),
+          this.firstMonomer,
+        ),
+      );
+    } else {
+      this.fillSubChains(
+        getNextMonomerInChain(
+          this.lastNode?.lastMonomerInNode,
+          this.firstMonomer,
+        ),
+      );
```

Comment:
I believe we need to keep only this part

Reply:
fixed